ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+                            handles=[legend_data_point,
+                                     (legend_hp, legend_hl), legend_fill],
+                            labels=["Data points", highlight_label, ci_label],
+                            loc="upper center",
+                            handler_map={
+                                tuple: mpl.legend_handler.HandlerTuple(ndivide=None)},
+                        )
+                    else:
+                        # Legend icon for the model mean function
+                        legend_mean = mpl.lines.Line2D(
+                            [],
+                            [],
+                            linestyle="-",
+                            color="red",
+                            marker="",
+                            lw=1,
+                        )
+                        ax[0][-1].legend(
+                            handles=[legend_data_point,
+                                     (legend_hp, legend_hl), legend_mean],
+                            labels=["Data points", highlight_label,
+                                    "Model mean function"],
+                            loc="upper center",
+                            handler_map={
+                                tuple: mpl.legend_handler.HandlerTuple(ndivide=None)},
+                        )
+
+    if usepartialdependence:
+        ylabel = "Partial dependence"
+    else:
+        ylabel = "Dependence"
+
+    return _format_scatter_plot_axes(
+        ax, space, ylabel=ylabel, dim_labels=dimensions
+    )
+
+
+def find_x_vals(pars, result, expected_minimum_samples):
+    # Here we define the values for which to plot the red dot (2d plot) and
+    # the red dotted line (1d plot). These same values will be used for
+    # evaluating the plots when calculating dependence. (Unless partial
+    # dependence is to be used instead).
+    if isinstance(pars, str):
+        if pars == "result":
+            # Using the best observed result
+            x_vals = result.x
+        elif pars == "expected_minimum":
+            # Do a gradient based minimum search using scipys own minimizer
+            if expected_minimum_samples:
+                # If a value for expected_minimum_samples has been parsed
+                num_random_starts = expected_minimum_samples
+            else:
+                # Use standard of 20 random starting points
+                num_random_starts = 20
+            x_vals, _ = expected_minimum(
+                result, n_random_starts=num_random_starts)
+        elif pars == "expected_minimum_random":
+            # Do a minimum search by evaluating the function with n_samples
+            # sample values
+            if expected_minimum_samples:
+                # If a value for expected_minimum_samples has been parsed
+                num_random_starts = expected_minimum_samples
+            else:
+                # Use standard of 10^n_parameters. Note this becomes very slow
```

Comment:
The comment and the code disagree here. I'm fine with hardcoding 1E5 as a value.